### PR TITLE
User Can See List Of Events (BE)

### LINKED ISSUE #12: User can see list of events (BE)
As a user, when I visit '/events', I can see a list of all drag events w/ event name, location, time, date, drag performers (3 max), and image with a button to view more info and a button to RSVP. 

- [x] Create event, artist, event_artists, location table
- [x] Create seed files / populate tables with data
- [x] Create REST endpoint to get all events

### PR DESCRIPTION
**Pull request by javier-aguilar**
## What functionality does this accomplish?
- Create event, artist, event_artists and location table
- Create seed files to populate tables above with sample data
- Create REST endpoint to get all events (/api/events)
- Add excoveralls for coverage testing
- Update GitHub Actions for CI workflow

## Current Test Suite:  **Mix/ExUnit**
16 tests, 0 failures
### Test Coverage Percentage: [TOTAL]  50.4%
- [ ] No tests have been changed
- [x] Some tests have been changed: Added testing for events (previously no files for testing)
- [ ] All of the tests have been changed:
## Checklist:
- [x] I have reviewed my code
- [ ] I have annotated or commented my code, particularly in hard-to-understand areas
- [ ] I have fully tested my code
- [x] I have partially tested my code (please explain why): I'm only testing for the events controller as we are not (yet?) utilizing the locations and artists controller. This is why the percentage is lower than expected
## Related Issues:
- Closes #12, #17 
## Helpful Resources:
- https://dev.to/ricardoruwer/many-to-many-associations-in-elixir-and-phoenix-21pm
- https://github.com/parroty/excoveralls
- https://nickvernij.nl/blog/github-ci-elixir.html
- https://medium.com/madeawkward/testing-your-phoenix-elixir-postgres-app-with-github-ci-5f3ec9f38ee0

### CODE REVIEW

File: lib/drag_time_api_web/controllers/event_controller.ex [hunk 1]
```diff
@@ -0,0 +1,53 @@
+defmodule DragTimeApiWeb.EventController do
+  use DragTimeApiWeb, :controller
+
+  alias DragTimeApi.Repo
+  alias DragTimeApi.Events
+  alias DragTimeApi.Events.Event
+
+  action_fallback DragTimeApiWeb.FallbackController
+
```
Comment (collinkallery) [inline lib/drag_time_api_web/controllers/event_controller.ex:9]: What is this grouping of files? Controllers?

**Review by collinkallery — APPROVED** [review]
Looks good so far - would like to hop on a call with y'all today to go over high-level what you've done here. How has it been working in Phoenix?

File: lib/drag_time_api_web/controllers/event_controller.ex (hunk 1 above)
Comment (javier-aguilar) [reply lib/drag_time_api_web/controllers/event_controller.ex:9]: The use DragTimeApiWeb is telling the controller to use the DragTimeApiWeb module. 

The alias section is a reference to other used modules within the controller. Setting alias DragTimeApi.Repo translates to Repo. So you don't have to call DragTimeApi.Repo within the controller, you can just call Repo.

**Comment by javier-aguilar** [comment]
Yeah sounds good. So far Phoenix hasn't been terribly difficult to work with. There's just not that much documentation or guides/tutorials for better understanding. But it's been working out.

File: lib/drag_time_api/artists.ex [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  @moduledoc """
+  The Artists context.
+  """
+
+  import Ecto.Query, warn: false
+  alias DragTimeApi.Repo
+
+  alias DragTimeApi.Artists.Artist
+
+  @doc """
```
Comment (atkinson-daniel) [inline lib/drag_time_api/artists.ex:11]: Nice use of inline documentation

File: lib/drag_time_api_web/controllers/event_controller.ex [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  def index(conn, _params) do
+    events = Repo.all(Event)
+      |> Repo.preload(:artists)
```
Comment (atkinson-daniel) [inline lib/drag_time_api_web/controllers/event_controller.ex:12]: Awesome, so I'm guessing we got the many to many association to work and render the artist and location info?

**Review by atkinson-daniel — APPROVED** [review]
Looks great - there's a lot of methods that Phoenix created on our behalf, so towards the end of the project we should remove the controller actions that we don't need.

File: lib/drag_time_api_web/controllers/event_controller.ex (hunk 3 above)
Comment (javier-aguilar) [reply lib/drag_time_api_web/controllers/event_controller.ex:12]: Yep!!

**Review by KCWill — COMMENTED** [review]
Nice work, looking forward to digging more into this later today!

**Pull request merged**